### DISCUSSION TITLE
cocoapods

### DISCUSSION THREAD

**Discussion by asnagni** [axmolengine/axmol#1162] [Q&A]
**Is your feature request related to a problem? Please describe.**
Yes, the problem is that we are trying to migrate from cocos2d-x to axmol. In the process, we are also trying to move the cocoapods integration that was done for cocos2d-x to axmol. For example we have firebase integrated into our project with the assistance of cocoapods.

We are having some issues implementing this process in axmol. Would have any documentation or steps on how to integrate cocoapods to axmol project?

**Describe the solution you'd like**
A documentation or steps describing how to integrate cocoapods with an axmol project

**Describe alternatives you've considered**
none

**Additional context**
none

**Comment by rh101** [comment]
What version of Cocos2d-x does your project use?

Others have asked this same question many times on the Cocos2d-x forum (specifically for v3/CMake, and V4/CMake), and as far as I know, there's never been a solution to this.  

If your project on Cocos2d-x has cocoapods integrated in some way with CMake, and all you are trying to do is migrate it to Axmol, then there shouldn't be anything in Axmol that is stopping you from doing this.  Also, if you really do have cocoapods working with CMake, then you're welcome to share how you integrated it to help others, and at the same time get more eyes looking at the problem you're having porting it over to Axmol.

Regardless of the above, this isn't an Axmol issue, but rather an issue with CMake support of cocoapods. 

Just a side note, Firebase and any other library can be integrated without cocoapods (I've personally done this with Firebase Crashlytics).

**Comment by mmokona** [comment]
Greetings. Try to do this:
1) Into CMakeLists.txt, add: 
`if(APPLE)
set(CMAKE_XCODE_ATTRIBUTE_FRAMEWORK_SEARCH_PATHS "${CMAKE_CURRENT_SOURCE_DIR}/build/Pods")
endif()`
2) Generate XCode project as written in documentation. 
3) Copy Podfile to ./build folder. Install Pods
4) Open project **Workspace Settings**, and change Derived Data path to ./build folder
![Untitled](https://github.com/axmolengine/axmol/assets/34329988/c156b5ec-7531-4aa4-8658-cd83f714a16c)
